metricbeat/module/postgresql/replication/replication.go

```diff
@@ -0,0 +1,72 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package replication
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/module/postgresql"
+
+	// Register postgresql as the sql driver
+	_ "github.com/lib/pq"
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	mb.Registry.MustAddMetricSet("postgresql", "replication", New,
+		mb.WithHostParser(postgresql.ParseURL),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet type defines all fields of the MetricSet.
+type MetricSet struct {
+	*postgresql.MetricSet
+}
+
+// New create a new instance of the postgresql replication MetricSet.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	ms, err := postgresql.NewMetricSet(base)
+	if err != nil {
+		return nil, err
+	}
+	return &MetricSet{MetricSet: ms}, nil
+}
+
+// Fetch method implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
+	ctx := context.Background()
+	results, err := m.QueryStats(ctx, "SELECT EXTRACT(EPOCH FROM write_lag)::INT AS write_lag, EXTRACT(EPOCH FROM flush_lag)::INT AS flush_lag, EXTRACT(EPOCH FROM replay_lag)::INT AS replay_lag, pid, usesysid, usename, application_name, client_addr, client_hostname, client_port, backend_start, backend_xmin, state, sent_lsn, write_lsn, replay_lsn, sync_priority, sync_state FROM pg_stat_replication;")
```

Comment:
This query will now extract the lags in epoch time with datatype as int?
And then you are doing a  .unit conversion using the new conversion function in data.go?

Am, I reading it right ?